src/views/Summary.js

```diff
@@ -15,25 +16,45 @@ function Summary({ selectedMonth, filteredExpenses }) {
     setTotalExpenses(expenseSum);
   }, [filteredExpenses]);
 
+  // Calculate spending rate deviation
+  useEffect(() => {
+    const date = new Date(selectedMonth);
+
+    // Get last day of selected month
+    const daysInMonth = new Date(
+      date.getFullYear(),
+      date.getMonth() + 1,
+      0
+    ).getDate();
+
+    // Set day equal to today's date or last day of the month
+    const dayOfMonth =
+      selectedMonth === currentMonth ? new Date().getDate() : daysInMonth;
+
+    const targetRate = totalBudget / daysInMonth;
+    const actualRate = totalExpenses / dayOfMonth;
+    const deviation = (actualRate - targetRate) / targetRate;
+    setSpendingRateDeviation(deviation);
+  }, [currentMonth, selectedMonth, totalBudget, totalExpenses]);
```

Comment:
the calculation is nice an clear. I am a little itchi however because the function is not named (we only see `useEffect`). howevery, you have used a comment, which makes it clearer. maybe we could do something like this:

```js
useEffect(updateExpenseTotal);
useEffect(updateSpendingRateHighlight);
```

what do you think?

Reply:
Sure thing. However, we'll still need to define those functions inside their respective calls to `useEffect` because of the dependency list issue, so we'd get something like this:

```
useEffect(() => {
  function updateSpendingRateDeviation() {
    // ...
    setSpendingRateDeviation(deviation);
  }
	
  updateSpendingRateDeviation();
}, [currentMonth, selectedMonth, totalBudget, totalExpenses]);
```

Reply:
ah, na. this is even worse :)

bloody dependency issues ;)

i think i already forgot what exactly the dependency issue was in useEffect. can you shoot me a link?

Reply:
**Conditionally firing side-effects:** https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect
**Is it safe to omit functions from dependency list**: https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

I suppose if we wanted to we could do away with the dependency list altogether, and then the side-effect would fire each time the component re-renders (instead of waiting for its dependencies to change).

Reply:
Okay, I can confirm that everything still works fine if we get rid of the dependency list and define two functions `updateTotalExpenses()` and `updateSpendingRateDeviation()` outside of `useEffect`. But as I mentioned above, these side-effects will fire every time `<Summary />` renders, regardless of whether or not their dependencies change.

Reply:
thanks for the link it was a great resource. I must say, after reading the article (again), I am still a little confused why React must be so complicated at times...

anyway, thanks again and also thanks for trying to change it.